js/index.js

```diff
@@ -2,128 +2,798 @@
-          var input = readInput('#input');
-          var output = parsedSchema.encode(input);
-          $('#output').val(bufferToStr(output));
-          clearErrors();
+          var input = readInput();
+          var output = window.type.toBuffer(input);
+          var outputStr = output.toString('hex');
+          setOutputText(outputStr);
+          eventObj.trigger('update-url', {'record' : outputStr});
+          eventObj.trigger('valid-output');
+          eventObj.trigger('valid-input');
         }catch(err) {
-          showError(decodedErrorElement, err);
+          eventObj.trigger('invalid-input', err);
         }
-      } else {
-        showError(decodedErrorElement, 'No valid schema found!');
       }
-     
-    });
+    }
 
-     $('#decode').click(function() {
-      if (parsedSchema) {
-        var input = readBuffer('#output');
+    function decode(rawInput) {
+      if (window.type) {
         try {
-          var decoded = parsedSchema.decode(input);
-          decoded = JSON.stringify(decoded, null, 2);
-          $('#input').val(decoded);
-          clearErrors();
+          var input = rawInput ? readBuffer(rawInput) : readBuffer();
+          var decoded = window.type.fromBuffer(input);
+          var decodedStr = window.type.toString(decoded);
+          setInputText(decodedStr);
+          eventObj.trigger('re-instrument');
+          eventObj.trigger('valid-input');
+          eventObj.trigger('valid-output');
         }catch(err) {
-          showError(encodedErrorElement,err);
+          eventObj.trigger('invalid-output', err);
         }
+      }
+    }
+
+    function showError(errorElem, msg) {
+      errorElem.text(msg);
+      errorElem.removeClass('-hidden-');
+    };
+
+    function hideError(errorElem, validElem) {
+      errorElem.text("");
+      errorElem.addClass('-hidden-');
+      validElem.show('slow').delay(500).hide('slow');
+    }
+    
+    function clearText(element) {
+      element.text('');
+    }
+    /* If the schema is pasted with proper json formats, simply json.parse wouldn't work.*/
+    function readSchemaFromInput() {
+      var trimmedInput = $.trim($('#schema').text()).replace(/\s/g, "");
+      return JSON.parse(trimmedInput);
+    }
+
+    function readInput() {
+      var rawInput = $.trim($(inputElement).text());
+      if(!!window.type) {
+        return window.type.fromString(rawInput);
       } else {
-        showError(encodedErrorElement, 'No valid schema found!');
+        return JSON.parse(rawInput);
       }
-    });
+    }
+    /*Used for decoding.
+    *Read the text represented as space-seperated hex numbers in elementId
+    *and construct a Buffer object*/
+    function readBuffer(str) {
+      var rawInput = str || $.trim(outputElement.text());
+      var hexArray = rawInput.split(/[\s,]+/);
+      var i;
+      var size = hexArray.length;
+      var buffer = [];
+      for (i =0; i < size; i++){
+        buffer.push(new Buffer(hexArray[i], 'hex'));
```

Comment:
Rather than split and combine (which requires the creation of multiple buffers), would it work to remove the commas and whitespace from the original input and pass the result directly?